Add steps for Frida snippet generation from JADX (#2331)
OWASP/owasp-mastg: Document/0x08a-Testing-Tools.md

The above script calls `Java.perform` to make sure that your code gets executed in the context of the Java VM. It instantiates a wrapper for the `android.app.Activity` class via `Java.use` and overwrites the `onResume` function. The new `onResume` function implementation prints a notice to the console and calls the original `onResume` method by invoking `this.onResume` every time an activity is resumed in the app.

> The [JADX decompiler](https://github.com/skylot/jadx) can generate Frida snippets through it's graphical code browser which print method entry and exit parameters via a `Java.use` call. To use this feature, open the APK or DEX with `jadx-gui`, browse to the target method, right click the method name, and select "Copy as frida snippet (f)". For example using the MASTG `UnCrackable-Level1.apk` app:

Reviewer: ```suggestion
The [JADX decompiler](#jadx) (v1.3.3 and above) can generate Frida snippets through its graphical code browser. To use this feature, open the APK or DEX with `jadx-gui`, browse to the target method, right click the method name, and select "Copy as frida snippet (f)". For example using the MASTG [UnCrackable App for Android Level 1](0x08b-Reference-Apps.md#android-uncrackable-l1):
```

<img src="Images/Chapters/0x08a/jadx_copy_frida_snippet.png" width="100%" />

> The above steps place the following output in pasteboard to be pasted in a `.js` file and fed into `frida -U -l`.

Reviewer: ```suggestion
The above steps place the following output in the pasteboard, which you can then paste in a JavaScript file and feed into `frida -U -l`.
```

> ```java

Reviewer: ```suggestion
```javascript
```

let a = Java.use("sg.vantagepoint.a.a");
a["a"].implementation = function (bArr, bArr2) {
    console.log('a is called' + ', ' + 'bArr: ' + bArr + ', ' + 'bArr2: ' + bArr2);
    let ret = this.a(bArr, bArr2);
    console.log('a ret value is ' + ret);
    return ret;
};
```

> The above code generated by `jadx-gui` hooks the `a` method within the `sg.vantagepoint.a.a` class and logs it's entry parameters and return values.

Reviewer: ```suggestion
The above code hooks the `a` method within the `sg.vantagepoint.a.a` class and logs its input parameters and return values.
```
